src/components/Navigation.vue

```diff
@@ -106,15 +108,29 @@ export default {
 	},
 	methods: {
 		onNewMessage() {
-			// FIXME: assumes that we're on the 'message' route already
-			this.$router.push({
-				name: 'message',
-				params: {
-					accountId: this.$route.params.accountId,
-					folderId: this.$route.params.folderId,
-					messageUid: 'new',
-				},
-			})
+			const accountId = this.$route.params.accountId || this.$store.getters.accounts[0].id
+
+			// FIXME: this assumes that there's at least one folder
+			const folderId = this.$route.params.folderId || this.$store.getters.getFolders(accountId)[0].id
+			if (this.$router.currentRoute.name === 'message' && this.$router.currentRoute.params.messageUid === 'new') {
+				// If we already show the composer, navigating to it would be pointless (and doesn't work)
+				// instead trigger an event to reset the composer
+				this.$root.$emit('newMessage')
+				return
+			}
+
+			this.$router
+				.push({
+					name: 'message',
+					params: {
+						accountId,
+						folderId,
+						messageUid: 'new',
```

Comment:
Needs adjustment for/after https://github.com/nextcloud/mail/pull/2706/files#diff-03f4e6dc240c55861fa235e27445d65bR86

Reply:
Should be adjusted